tests/test_clap_cost.py

```diff
@@ -1,43 +1,103 @@
+
+global_cost_3 = np.array([[6, 5, 6],
+                          [5, 6, 9],
+                          [6, 7, 5],
+                          [8, 11, 11]])
+
+global_cost_4 = np.array([[np.Inf, np.Inf, np.Inf, np.Inf,     47],
+                          [    47, np.Inf,     49, np.Inf, np.Inf],
+                          [    49,     47, np.Inf, np.Inf, np.Inf],
+                          [np.Inf,     49,     47,     54, np.Inf]])
+
+global_cost_5 = np.array([[6, 5, 6, np.Inf],
+                          [5, 6, 9, np.Inf],
+                          [6, 7, 5, np.Inf]])
+
+global_cost_6 = np.array([[6, 5, 6],
+                          [5, 6, 9],
+                          [np.Inf, np.Inf, np.Inf],
+                          [6, 7, 5],
+                          [8, 11, 11]])
 # fmt: on
 
 
-@pytest.mark.parametrize(
-    "i, j, costs, expected",
-    [
-        (0, 0, costs1, 6),
-        (0, 1, costs1, 5),
-        (0, 2, costs1, 6),
-        (1, 0, costs1, 5),
-        (1, 1, costs1, 6),
-        (1, 2, costs1, 9),
-        (2, 0, costs1, 6),
-        (2, 1, costs1, 7),
-        (2, 2, costs1, 5),
-        (0, 3, costs2, 8),
-        (1, 3, costs2, 10),
-        (2, 3, costs2, 11),
-        (3, 0, costs3, 8),
-        (3, 1, costs3, 11),
-        (3, 2, costs3, 11),
-    ],
-)
-def test_clap_cost(i, j, costs, expected):
-    """Verify clap.cost works on small examples computable by hand."""
-    assert clap.cost(i, j, costs) == expected
+class TestClap:
+    @pytest.mark.parametrize(
+        "i, j, cost_matrix, expected",
+        [
+            (0, 0, cost_matrix_1, 6),
+            (0, 1, cost_matrix_1, 5),
+            (0, 2, cost_matrix_1, 6),
+            (1, 0, cost_matrix_1, 5),
+            (1, 1, cost_matrix_1, 6),
+            (1, 2, cost_matrix_1, 9),
+            (2, 0, cost_matrix_1, 6),
+            (2, 1, cost_matrix_1, 7),
+            (2, 2, cost_matrix_1, 5),
+            (0, 3, cost_matrix_2, 8),
+            (1, 3, cost_matrix_2, 10),
+            (2, 3, cost_matrix_2, 11),
+            (3, 0, cost_matrix_3, 8),
+            (3, 1, cost_matrix_3, 11),
+            (3, 2, cost_matrix_3, 11),
+        ],
+    )
+    def test_clap_cost(self, i, j, cost_matrix, expected):
+        """Verify clap.cost works on small examples computable by hand."""
+        assert clap.cost(i, j, cost_matrix) == expected
+
+    @pytest.mark.parametrize(
+        "cost_matrix, expected_global_costs",
+        [
+            (cost_matrix_1, global_cost_1),
+            (cost_matrix_2, global_cost_2),
+            (cost_matrix_3, global_cost_3),
+            (cost_matrix_4, global_cost_4),
+            (cost_matrix_5, global_cost_5),
+            (cost_matrix_6, global_cost_6),
+        ],
+    )
+    def test_clap_costs(self, cost_matrix, expected_global_costs):
+        """Verify clap.constrained_lsap_costs works on small examples."""
+        assert np.all(clap.costs(cost_matrix) == expected_global_costs)
```

Comment:
It would be preferable to use 
```python
assert clap.costs(cost_matrix).tolist() == expected_global_costs.tolist()
```
since the output will be more useful when one of the tests fails. i.e. it will tell you exactly which elements were not equal. There might be alternatives such as [pytest-allclose](https://www.nengo.ai/pytest-allclose/) if you are interested in looking for a nicer way to handle this.